modules/natgw/variables.tf

```diff
@@ -29,84 +19,149 @@ variable "tags" {
+  Triggers creation of a NAT Gateway when set to `true`.
+  
+  Set it to `false` to source an existing resource. In this 'mode' the module will only bind an existing NAT Gateway to specified
+  subnets.
+  EOF
+  default     = true
+  type        = bool
+}
+
 variable "zone" {
   description = <<-EOF
-  Controls if the NAT Gateway will be bound to a specific zone or not. This is a string with the zone number or `null`. Only for newly created resources.
+  Controls whether the NAT Gateway will be bound to a specific zone or not. This is a string with the zone number or `null`. Only
+  for newly created resources.
 
-  NAT Gateway is not zone-redundant. It is a zonal resource. It means that it's always deployed in a zone. It's up to the user to decide if a zone will be specified during resource deployment or if Azure will take that decision for the user. 
-  Keep in mind that regardless of the fact that NAT Gateway is placed in a specific zone it can serve traffic for resources in all zones. But if that zone becomes unavailable resources in other zones will loose internet connectivity. 
+  NAT Gateway is not zone-redundant. It is a zonal resource. It means that it's always deployed in a zone. It's up to the user to
+  decide if a zone will be specified during resource deployment or if Azure will take that decision for the user. Keep in mind
+  that regardless of the fact that NAT Gateway is placed in a specific zone it can serve traffic for resources in all zones. But
+  if that zone becomes unavailable, resources in other zones will lose internet connectivity.
 
   For design considerations, limitation and examples of zone-resiliency architecture please refer to [Microsoft documentation](https://learn.microsoft.com/en-us/azure/virtual-network/nat-gateway/nat-availability-zones).
   EOF
   default     = null
   type        = string
+  validation {
+    condition     = (var.zone == null || can(regex("^[1-3]$", var.zone)))
+    error_message = "The `zone` variable should have value of either: \"1\", \"2\" or \"3\"."
+  }
 }
 
 variable "idle_timeout" {
-  description = "Connection IDLE timeout in minutes. Only for newly created resources."
-  default     = null
+  description = "Connection IDLE timeout in minutes (up to 120, by default 4). Only for newly created resources."
+  default     = 4
   type        = number
+  validation {
+    condition     = (var.idle_timeout >= 1 && var.idle_timeout <= 120)
+    error_message = "The `idle_timeout` variable should be a number between 1 and 120."
+  }
 }
 
 variable "subnet_ids" {
-  description = "A map of subnet IDs what will be bound with this NAT Gateway. Value is the subnet ID, key value does not matter but should be unique, typically it can be a subnet name."
+  description = <<-EOF
+  A map of subnet IDs what will be bound with this NAT Gateway. Value is the subnet ID, key value does not matter but should be
+  unique, typically it can be a subnet name.
+  EOF
   type        = map(string)
```

Comment:
the point of 130 columns is just readability, so whatever you do with the text, just keep in mind that it should be easily digestible :) in all 4 places, `variables.tf`, `README.md`, terraform registry, pan.dev

just an example:

```hcl
description = <<-EOF
  A map of subnet IDs what will be bound with this NAT Gateway.

  Value is the subnet ID, key value does not matter but should be unique.
  Typically it can be a subnet name.

  EOF
```